lua/lazyvim/plugins/extras/lang/yaml.lua

```diff
@@ -0,0 +1,30 @@
+return {
+
+  -- add yaml specific modules to treesitter
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "yaml" })
+      end
+    end,
+  },
+
+  -- correctly setup lspconfig
+  {
+    "neovim/nvim-lspconfig",
+    dependencies = {
+      "someone-stole-my-name/yaml-companion.nvim",
+      version = false, -- last release is way too old
+      config = function(_, opts)
+        require("telescope").load_extension("yaml_schema")
+      end,
+    },
+    opts = {
+      -- make sure mason installs the server
+      servers = {
+        yamlls = {}
```

Comment:
Should use `SchemaStore` here:

```lua
      yamlls = {
        ---@param new_config lspconfig.options.yamlls
        on_new_config = function(new_config)
          new_config.settings.yaml.schemas = new_config.settings.yaml.schemas or {}
          vim.list_extend(new_config.settings.yaml.schemas, require("schemastore").yaml.schemas())
        end,
      },
```

Reply:
also turn telemetry off by default:
```lua
{ telemetry = { enabled = false } },
```

Reply:
its the default for the `yaml-companion...`: https://github.com/someone-stole-my-name/yaml-companion.nvim#%EF%B8%8F--configuration

Reply:
`yaml-companion`'s builtin SchemaStore integration does not provides local cache on the catalog and may possibly cause slowdown on file matching. `SchemaStore.nvim` does auto-generate with CI: <https://github.com/b0o/SchemaStore.nvim/blob/main/lua/schemastore/catalog.lua>

Also this brings better consistency with <https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/extras/lang/json.lua>.

Reply:
Thanks, updated according to README at https://github.com/b0o/SchemaStore.nvim